### PR TITLE
My expansion of the documentation

### PR DESCRIPTION
**Pull request by astei**
TODO:

- [x] Add more detail to the serializers
- [x] Explain more about what audiences are and what problems they solve
- [ ] Clean up and professionalize the MiniMessage documentation
- [x] Finish explaining platforms and explore platforms that natively support adventure
  - [x] Explore ViaVersion support
- [ ] Migration guide from BungeeCord chat component API and text 3.x
- [ ] Stretch goals:
  - [ ] Boss bars
  - [ ] Titles

### CODE REVIEW

File: source/serializer/gson.rst [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+As explained in the name, the Gson serializer uses the Gson library to serialize the
+chat components to JSON.
+
+Usage
+-----
+
+The Gson serializer is accessed using the ``GsonComponentSerializer`` class. You can
+use ``GsonComponentSerializer.gson()`` for a default instance that supports RGB components
+(only Minecraft 1.16 and above will be able to deserialize these components) or use
+``GsonComponentSerializer.colorDownsamplingGson()`` which downsamples colors to the
```
Comment (zml2008) [inline source/serializer/gson.rst:19]: Should probably mention the `AudienceProvider.gsonSerializer()` method here -- it's a fairly recent addition, that gets the gson serializer most appropriate for the running server version.

**Comment by astei** [comment]
Some more details to cover:

- [ ] What do we need to cover for serializers? (I think I've covered the bare basics.)
- [ ] Do we need to explain the platforms some more? What should be covered aside from the artifact to use and some basic usage guidelines?
- [ ] What's the best way to discuss audiences?

File: source/migration/bungeecord-chat-api.rst [hunk 2]
```diff
@@ -5,17 +5,90 @@ Migrating from the BungeeCord Chat API
...
+
+Adventure's equivalents for ``ChatColor`` are split over three types:
+
+  * Formatting types (such as ``BOLD`` or ``ITALIC``) are in ``TextDecoration``, and can be set
+    on a component or a style with the ``decoration`` method. Decorations also use a tristate to
+    specify if they are enabled, disabled, or not set (in which case the component inherits the
+    setting from its parent component).
+  * Named colors (also called the legacy Mojang color codes) now exist in the ``NamedTextColor``
+    class.
+  * RGB colors are constructed using the ``TextColor.of()`` method.
```
Comment (kashike) [inline source/migration/bungeecord-chat-api.rst:43]:
```suggestion
  * RGB colors are constructed using the ``TextColor.of()`` methods.
```
More than one method exists.

File: source/migration/bungeecord-chat-api.rst [hunk 3]
```diff
@@ -35,6 +108,25 @@ becomes this Adventure equivalent:
...
+    .bold(true)
+    .append(" world")
+    .build()
+
+becomes
+
+.. code:: java
+
+  Style style = Style.of(NamedTextColor.GOLD).decoration(TextDecoration.BOLD, TextDecoration.State.TRUE);
+  TextComponent.builder("")
```
Comment (kashike) [inline source/migration/bungeecord-chat-api.rst:126]: Content defaults to an empty string already.
```suggestion
  TextComponent.builder()
```

File: source/migration/bungeecord-chat-api.rst [hunk 4]
```diff
@@ -35,6 +108,25 @@ becomes this Adventure equivalent:
...
+    .color(ChatColor.GOLD)
+    .bold(true)
+    .append(" world")
+    .build()
+
+becomes
+
+.. code:: java
+
+  Style style = Style.of(NamedTextColor.GOLD).decoration(TextDecoration.BOLD, TextDecoration.State.TRUE);
```
Comment (kashike) [inline source/migration/bungeecord-chat-api.rst:125]: Can be simpler,
```suggestion
  Style style = Style.of(NamedTextColor.GOLD, TextDecoration.BOLD);
```

File: source/migration/text-3.x.rst [hunk 5]
```diff
@@ -6,10 +6,39 @@ Adventure is an evolution of the text 3.x API. If you've worked with
...
+In text 3.x, you would deserialize a component that used a color code prefix that
+differed from the section symbol normally used by using ``LegacyComponentSerializer.legacy().deserialize(string, altChar)``.
+In Adventure, the API to use is ``LegacyComponentSerializer.legacy(altChar).deserialize(string)``.
+
+``TextColor`` renamed to ``NamedTextColor``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to accomodate the new RGB colors introduced in 1.16, all the named text colors
+were moved to the ``NamedTextColor`` class. References to the old ``TextColor`` class
+should be updated to refer to ``TextColor``.
```
Comment (kashike) [inline source/migration/text-3.x.rst:36]: Replace `TextColor` with `TextColor`? That does not make much sense?

**Comment by astei** [comment]
Not abandoned, just caught up in other projects (such as the new Velocity website).

**Pull request merged**